core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -132,7 +132,8 @@ public static ScalarFunction createColumn(int position, CharSequence name, Recor
         }
     }
 
-    public Function createBindVariable(int position, CharSequence name) throws SqlException {
+    public Function createBindVariable(SqlExecutionContext sqlExecutionContext, int position, CharSequence name) throws SqlException {
+        this.sqlExecutionContext = sqlExecutionContext;
```

Comment:
nit: we could also make `FunctionParser` mutable and set `sqlExecutionContext` to null in its `clear()` method. Then `SqlCodeGenerator#clear()` could clear the parser. This way missing initialization will immediately manifest itself through NPEs.

Reply:
thank you Andrey, makes sense https://github.com/questdb/questdb/pull/1742/commits/2df5cfb39e1d753e934c240203b96cea1e2c8732